src/main/java/com/exadel/sober/controllers/UserController.java

```diff
@@ -1,117 +1,57 @@
-
-    @PostMapping(path="/add")
-    public @ResponseBody String addNewUser(@RequestBody User newUser) {
-        User userForSave = newUser;
-        userForSave.setPassword(get_SHA_512_SecurePassword(newUser.getPassword()));
-        userRepository.save(newUser);
-        return "User " +newUser.getName()+ " registration successful";
+    private UserService userService;
+    public UserController(UserRepository userRepository,
+                          UserService userService) {
+        this.userRepository = userRepository;
+        this.userService = userService;
     }
 
-    @PostMapping(path="/login")
-    public @ResponseBody String LoginUser(@RequestBody User loginUser) {
-        boolean valid = false;
-        if (userRepository.existsByName(loginUser.getName())) {
-            User user = userRepository.findByName(loginUser.getName());
-            if (Objects.equals(user.getName(), loginUser.getName())
-                    && Objects.equals(user.getPassword(), get_SHA_512_SecurePassword(loginUser.getPassword()))) {
-              valid = true;
-            }
-        }
-        if (valid) {
-            return loginUser.getName();
-        }
-        else {
-            return "credential invalid";
-        }
-    }
-
-    @PostMapping(path="/delete")
-    public @ResponseBody String deleteNewUser(@RequestParam String name) {
-        //User n = userRepository.findByName(name);
-        //userRepository.delete(n);
-        return "User " +name+ " was deleted";
-    }
-
-    @PostMapping(path="/addPromise")
-    public @ResponseBody List<Promise> addPromiseByUserName (@RequestParam String userName, @RequestParam Integer addiction_id, @RequestParam Integer days) {
-        User user = userRepository.findByName(userName);
-        Addiction addiction = addictionRepository.findById(addiction_id).get();
-        Integer d = days;
-
-        Promise p = new Promise();
-        p.setAddictionId(addiction.getAddictionId());
-        p.setUserId(user.getUserId());
-        p.setName("i will not being take " + addiction.getName());
-        p.setExpiredDate(new Timestamp(System.currentTimeMillis() + d*24*3600*1000));
-        promiseRepository.save(p);
-        return userRepository.findByName(userName).getPromises();
+    @GetMapping()
+    public @ResponseBody List<User> getAllUsers() {
+        return userRepository.findAll();
     }
 
-    @GetMapping(path="/getPromises")
-    public @ResponseBody List<Promise> getPromiseByUserName (@RequestParam String userName) {
-
-        User user = userRepository.findByName(userName);
-        return user.getPromises();
+    @PostMapping()
+    public @ResponseBody User addNewUser(@RequestBody User newUser) {
+        return userService.addNewUser(newUser);
     }
 
-    @PostMapping(path="/deletePromise")
-    public @ResponseBody void deletePromiseByUserName (@RequestParam String name) {
+    @PostMapping(path="/login")
+    ResponseEntity<Object> LoginUser(@RequestBody User loginUser) {
```

Comment:
ResponceEntity is very good point. Please provide it for all controller responcess